sources/whippet/src/broker.rs

```diff
@@ -0,0 +1,138 @@
+}
+
+/// Manages the lifecycle of the spawned broker
+pub(crate) struct BrokerManager {
+    child: BrokerChild,
+    controller: UnixStream,
+    policy: DbusPolicy,
+}
+
+impl BrokerManager {
+    pub(crate) async fn build_with(broker_path: &str, policy: DbusPolicy) -> Result<Self> {
+        debug!("Creating BrokerManager with broker path: {}", broker_path);
+
+        debug!("Creating controller socket pair");
+        let (controller, broker_end) = socket::controller_pair()?;
+
+        debug!("Creating journal socket");
+        let journal_socket = socket::journal_socket()?;
+
+        // Transfer ownwership of the sockets to the child process, so that they are cleaned up
+        // automatically once the child goes out of scope (no extra drop calls required)
+        let child = BrokerChild::new(broker_path, broker_end, journal_socket)?;
+
+        Ok(Self {
+            child,
+            controller,
+            policy,
+        })
+    }
+
+    /// Spawn the broker and transfer the systemd socket and policy
+    pub(crate) async fn run(&mut self) -> Result<()> {
+        // Start the broker in the background but don't block on it being ready, as the AddListener
+        // call will block until the broker is ready to receive calls in the controller socket
+        self.child.run()?;
+        debug!("Adding listener with parsed policy");
+        self.add_listener().await?;
+        info!("Policy applied successfully");
+        // Notify the broker is ready once the policy and the sockets were passed to the broker.
+        // This deviates slightly from the dbus-launcher as it notifies systemd right after the
+        // configuration is loaded
+        self.notify_systemd_ready()?;
+        info!("Notified systemd that service is ready");
+
+        // Wait forenver until signaled to stop or the dbus-broker is killed
+        self.child.wait().await
+    }
+
+    /// Call AddListener method with policy
+    async fn add_listener(&self) -> Result<()> {
+        debug!("Starting AddListener call");
+
+        // Get systemd listener socket
+        debug!("Getting systemd listener socket");
+        let listener_fd = socket::listener_socket()?;
+        info!("Using listener socket FD: {}", listener_fd);
+
+        // Create object path and fd handle
+        let object_path = ObjectPath::try_from("/org/bus1/DBus/Listener/0")
+            .context(error::ParseObjectPathSnafu)?;
+        // The listener file descriptor technically isn't owned by whippet. Wrap the file
+        // descriptor around a BorrowedFd to prevent the file descriptor from being closed once the
+        // broker goes out of scope
+        let fd_handle = Fd::from(unsafe { BorrowedFd::borrow_raw(listener_fd as i32) });
+
+        info!("Calling AddListener method");
+
+        // Wrap policy in variant (like launcher does with sd_bus_message_open_container(m, 'v', ...))
+        let policy_variant = Value::from(self.policy.clone());
```

Comment:
nit: avoid including type information in the variable name:
```suggestion
        let policy = Value::from(self.policy.clone());
```